src/test/ui/lint/lint-deref-nullptr.rs

```diff
@@ -0,0 +1,32 @@
+// test the deref_nullptr lint
+
+#![deny(deref_nullptr)]
+
+fn f() {
+    unsafe {
+        let a = 1;
+        let ub = *(a as *const i32);
+        let ub = *(0 as *const i32);
+        //~^ ERROR Dereferencing a null pointer causes undefined behavior
+        let ub = *core::ptr::null::<i32>();
+        //~^ ERROR Dereferencing a null pointer causes undefined behavior
+        let ub = *core::ptr::null_mut::<i32>();
+        //~^ ERROR Dereferencing a null pointer causes undefined behavior
+        let ub = *(core::ptr::null::<i16>() as *const i32);
+        //~^ ERROR Dereferencing a null pointer causes undefined behavior
+        let ub = *(core::ptr::null::<i16>() as *mut i32 as *mut usize as *const u8);
```

Comment:
Nice. :)
If I understand your lint code correctly, even `core::ptr::null::<i16>() as usize as *const u8` should be detected, right?

Reply:
Ah no, you are testing for `rustc_hir::TyKind::Ptr(_)`. Not a big deal, that does not seem like code anyone would write.

Reply:
for the moment no `core::ptr::null::<i16>() as usize as *const u8` is not detected but I can add that :)

Reply:
No need, that doesn't seem like something people would actually write.^^